src/windshaft-integration/legends/rule-to-category-legend-adapter.js

```diff
@@ -11,12 +11,23 @@ var isEveryBucketValid = function (rule) {
   });
 };
 
-var generateCategories = function (buckets) {
-  return _.map(buckets, function (bucket) {
-    return { label: bucket.filter.name, value: bucket.value };
+var generateCategories = function (bucketsColor, bucketsIcon) {
+  return _.map(bucketsColor, function (bucketColor) {
+    var bucketIcon = _.find(bucketsIcon, function (bucket) {
+      return bucket.filter.name === bucketColor.filter.name;
+    });
+    return {
+      title: bucketColor.filter.name,
+      icon: bucketIcon ? _extractURL(bucketIcon.value) : '',
+      color: bucketColor.value
+    };
   });
 };
 
+var _extractURL = function (str) {
+  return str.slice(5, -2);
```

Comment:
be wary of magic numbers, I understand why, but the source value not always come quoted, actually it doesn't appear quoted in the docs https://carto.com/docs/carto-engine/cartocss/properties/#uri

shouldn't we better match the url inside the `url()` rahter than rely on the position of the url in the string?

Reply:
Yeah, I was hoping for a better method to do this. Do you recommend regex for the url?